src/Microsoft.Deployment.DotNet.Releases/src/Product.cs

```diff
@@ -0,0 +1,212 @@
+        /// be checked to confirm whether a release is still supported. 
+        /// </summary>
+        /// <remarks>
+        /// The EOL dates are often published in advance, but there can be delays to updating the support phase in the published
+        /// data.
+        /// </remarks>
+        [JsonProperty(PropertyName = "support-phase")]
+        public SupportPhase SupportPhase
+        {
+            get;
+            private set;
+        } = SupportPhase.Unknown;
+
+        /// <summary>
+        /// Gets a collection of all releases associated with this <see cref="Product"/>.
+        /// </summary>
+        /// <returns>A collection of all releases for this product.</returns>
+        public async Task<ReadOnlyCollection<ProductRelease>> GetReleasesAsync()
+        {
+            return await GetReleasesAsync(ReleasesJson, this);
+        }
+
+        /// <summary>
+        /// Gets a collection of all releases associated with this <see cref="Product"/> using a file
+        /// containing the releases data.
+        /// </summary>
+        /// <param name="path">The path of the file containing the releases data.</param>
+        /// <param name="downloadLatest">When <see langword="true"/>, the latest copy of the releases data is used
+        /// if the online version is newer than the local file copy.</param>
+        /// <returns>A collection of releases associated with this <see cref="Product"/>.</returns>
+        public async Task<ReadOnlyCollection<ProductRelease>> GetReleasesAsync(string path, bool downloadLatest)
+        {
+            await Utils.GetLatestFileAsync(path, downloadLatest, ReleasesJson);
+
+            using (TextReader reader = File.OpenText(path))
+            {
+                return await GetReleasesAsync(reader, this);
+            }
+        }
+
+        /// <summary>
+        /// <see langword="true"/> if the support phase is <see cref="SupportPhase.EOL"/>
+        /// or the current date is greater than or equal to the EOL date of the product, 
+        /// <see langword="false"/> otherwise.
+        /// </summary>
+        /// <returns><see langword="true"/> if the product is out of support; <see langword="false"/> otherwise.</returns>
+        public bool IsOutOfSupport()
+        {
+            return SupportPhase == SupportPhase.EOL || EndOfLifeDate?.Date <= DateTime.Now.Date;
+        }
+
+        /// <summary>
+        /// Creates a new <see cref="ProductRelease"/> collection using the releases.json file pointed to
+        /// by the provided URL.
+        /// </summary>
+        /// <param name="address">The URL pointing to the releases.json file to use.</param>
+        /// <param name="product">The <see cref="Product"/> to link to the releases.</param>
+        /// <returns></returns>
+        public static async Task<ReadOnlyCollection<ProductRelease>> GetReleasesAsync(Uri address, Product product)
```

Comment:
The non-static one uses the default URL. The static one was intended if users want to mirror the releases.json files on their own server, e.g. an intranet.

Reply:
I understand that.  It is very strange to have a static method of Product that has a Product parameter.  What is the reason to not make it an instance method?

Reply:
The way I thought about it is that the instance method would use the default URLs and felt like good symmetry given the static method we have on the ProductCollection

Assuming you're hosting the files on a different server, the releases-index would likely point to local copies of the releases.json.

Maybe we should just remove it for now. The only other use case would be if you knew where a specific release lived and didn't care about the top level products, but only wanted to get the releases information.

Reply:
I see the merit in the scenario.  I see this as different than the ProductCollection static method because instance state is not required.

Another option is perhaps the ReleasesJson setter should be public.  For local copy scenarios, the user would tweak the ReleasesJson property before calling the instance GetReleasesAsync method.

@terrajobst, can you provide guidance here?

Reply:
The ReleasesJson property reflects the URL associated with the product as defined in the releases-index.json. Overriding that seems to defeat the main case of an application working its way through the ProductCollection->Product->Releases.

I'd prefer changing the static to an instance method instead of making the setter public

Reply:
@terrajobst this is the last issue we need to close on. Feel free to ping me if you need any detail.

Reply:
I agree wit @MichaelSimons. It seems more natural to have them regular instance overloads. It makes sense to me so see some with a URL and some without; this would indicate that URL is optional and if not specified a default is being used.

Reply:
This has now been addressed